slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -57,26 +57,49 @@ abstract class AbstractSourceCodeGenerator(model: m.Model)
       else compoundValue(types)
     }
 
+    
     val maxTupleSize = 22
+    def depth(numberOfColumns: Int) = (Math.log(numberOfColumns)/Math.log(maxTupleSize)).ceil.toInt
     //build and nest tuples if > 22 elements
-    def nestedTuple(values: Seq[String] , prefix: String): String = {
+    def nestedTuple(values: Seq[String], prefix: String): String = {
       def mkTuple[T](l: Seq[T], _prefix: String) = l.map{ _prefix + _ }.mkString("(" , ", " , ")")
       if(values.size <= maxTupleSize)
         mkTuple( values, "" )
       else 
-        mkTuple( values.grouped(maxTupleSize).toList.map {mkTuple(_,prefix)}, "" )
+      {
+        val d = depth(values.size)
+        if( d == 2 ) 
+          mkTuple( values.grouped(maxTupleSize).toList.map {mkTuple(_,prefix)}, "" )
+        else //support up to maxTupleSize^3 columns (10648 columns if maxTupleSize = 22) 
+          mkTuple( values.grouped(maxTupleSize*maxTupleSize).toList.map{l2 => mkTuple(l2.grouped(maxTupleSize).toList.map{ mkTuple(_, prefix) }, "") } , "")
+      }         
```

Comment:
can we solve this with recursion instead of hard coding a depth? (we may not need the depth function after all)
Somthing like `if(values.size / 22 == 0) values.mkString("(",", ",")") else values.grouped(values.size / 22).map(nestedTuple).mkString("(",", ",")"))`